lib/vagrant-service-manager/action/setup_network.rb

```diff
@@ -0,0 +1,37 @@
+module Vagrant
+  module ServiceManager
+    module Action
+      class SetupNetwork
+
+        def initialize(app, env)
+          @app = app
+          @machine = env[:machine]
+          @ui = env[:ui]
+        end
+
+        def call(env)
+          add_prerequisite_network if virtualbox? && !private_network_exists?
+          @app.call(env)
+        end
+
+        private
+
+        def virtualbox?
+          @machine.provider.instance_of?(VagrantPlugins::ProviderVirtualBox::Provider)
+        end
+
+        def private_network_exists?
+          @machine.config.vm.networks.any? { |type, _| type == :private_network }
+        end
+
+        def add_prerequisite_network
+          return unless virtualbox? && !private_network_exists?
+
+          @ui.info 'Virtualbox requires an additional private network. Adding it...'
```

Comment:
> As users run multiple instance of the Vagrantbox (usually two) 

Do they? why? What's the use case here?

> So if service-manager can configure the private networking with dhcp then adb-utils scripts can pick the IP address and use it for deploying OpenShift

I was not so much concerned about the provisioning step, but was more thinking about how the user knows the right IPs and URLs. If we have a "hard coded" default, users might have it easier to get started. On the other hand, with service-manager they can just run `vagrant service-manager env docker|openshift` and get the info they need. I guess that is how `docker-machine` works as well. 

Maybe we would still need a `vagrant service-manager ip` which returns the DHCP private network IP (independent of any specific service being enabled). Actually I quite like this idea. It's a step away from the default `10.1.2.2` approach we have been using. We might want to float this idea on a broader audience to get some feedback. WDYT?